src/rgw/rgw_sal_motr.cc

```diff
@@ -1376,11 +1411,25 @@ bool MotrObject::is_expired() {
 // Taken from rgw_rados.cc
 void MotrObject::gen_rand_obj_instance_name()
 {
-  enum {OBJ_INSTANCE_LEN = 32};
-  char buf[OBJ_INSTANCE_LEN + 1];
-
-  gen_rand_alphanumeric_no_underscore(store->ctx(), buf, OBJ_INSTANCE_LEN);
-  key.set_instance(buf);
+  // To list/store object versions in lexicographicaly sorted order,
+  // we are creating version-id based on timestamp value.
+  // If we generate any random string as version-id, list-object-version is
+  // not returning version-id in sorted order, and 'is-latest' flag value is
+  // true for multiple versions.
+  char buf[UUID_LEN + 1];
+  std::string version_id;
+  //TODO: Handle null version object case in PutObj operation.
+  // As the version ID timestamp is encoded in Base62, the maximum value
+  // for 8-characters is 62^8 - 1. This is the maximum time interval in ms.
+  constexpr uint64_t max_ts_count = 218340105584895;
+  using UnsignedMillis = std::chrono::duration<uint64_t, std::milli>;
+  const auto ms_since_epoch = std::chrono::time_point_cast<UnsignedMillis>(
+                              std::chrono::system_clock::now()).time_since_epoch().count();
+  uint64_t cur_time = max_ts_count - ms_since_epoch;
+  auto version_ts = base62_encode(cur_time, TS_LEN);
+  gen_rand_alphanumeric_no_underscore(store->ctx(), buf, UUID_LEN+1);
```

Comment:
why  UUID_LEN+1 ? and just UUID_LEN

Reply:
In this function, we need to pass this size parameter value as - required string size + 1.